jupyterhub/auth.py

```diff
@@ -37,6 +37,25 @@ class Authenticator(LoggingConfigurable):
     """Base class for implementing an authentication provider for JupyterHub"""
 
     db = Any()
+    
+    enable_auth_state = Bool(False, config=True,
+        help="""Enable persisting auth_state (if available).
+
+        auth_state will be encrypted and stored in the Hub's database.
+        This can include things like authentication tokens, etc.
+        to be passed to Spawners as environment variables.
+
+        Encrypting auth_state requires the cryptography package.
+
+        Additionally, the JUPYTERHUB_CRYPTO_KEY envirionment variable must
```

Comment:
> I suspect we might have other encryption keys in the future, and would not want to re-use the same key.

I was thinking the opposite - if we add any other encrypted data into the database, we *would* use the same key, wouldn't we? Is there value in having different keys encrypting different columns of the database? If that's wrong, I'll definitely change the name back to include AUTH_STATE.